cmd/mdatagen/templates/config.go.tmpl

```diff
@@ -10,6 +10,12 @@ import (
 )
 {{ if .Metrics -}}
 
+var MetricNames = []string{
```

Comment:
Good catch thanks, this was an accident

Reply:
Done in [a758241](https://github.com/open-telemetry/opentelemetry-collector/pull/10065/commits/a75824167ff8bb9038e84d5e7e9f39b08d8e1746)

I couldn't render directly in `expandPatternMap` because there could be either metrics or resource attributes passed in. I render both arrays more locally to their usage though instead of as globals.